From 66c004585109e69badd79b66665e8405f49fcaed Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Fri, 24 Nov 2017 18:52:54 +0800
Subject: [PATCH 2896/5242] MLK-16977-2 media: v4l_cap: don't support suspend
 when camera is running

commit  e663bcac87e95a472325b00eba13b791ffebddcb from
https://source.codeaurora.org/external/imx/linux-imx.git

Don't support suspend/resume by checking if camera is running.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-mipi-csi2_yav.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c b/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
index ac9a8c4..247b880 100644
--- a/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
+++ b/drivers/media/platform/imx8/mxc-mipi-csi2_yav.c
@@ -671,13 +671,14 @@ static int mipi_csi2_runtime_pm_suspend(struct device *dev)
 static int mipi_csi2_pm_suspend(struct device *dev)
 {
 	struct mxc_mipi_csi2_dev *csi2dev = dev_get_drvdata(dev);
-	struct v4l2_subdev *sd = &csi2dev->sd;
 
 	if (csi2dev->flags & MXC_MIPI_CSI2_PM_SUSPENDED)
 		return 0;
 
-	if (csi2dev->running)
-		mipi_csi2_s_stream(sd, false);
+	if (csi2dev->running) {
+		dev_warn(dev, "running, prevent entering suspend.\n");
+		return -EAGAIN;
+	}
 	mipi_csi2_clk_disable(csi2dev);
 	csi2dev->flags &= ~MXC_MIPI_CSI2_PM_POWERED;
 	csi2dev->flags |= MXC_MIPI_CSI2_PM_SUSPENDED;
@@ -688,7 +689,6 @@ static int mipi_csi2_pm_suspend(struct device *dev)
 static int mipi_csi2_pm_resume(struct device *dev)
 {
 	struct mxc_mipi_csi2_dev *csi2dev = dev_get_drvdata(dev);
-	struct v4l2_subdev *sd = &csi2dev->sd;
 	int ret;
 
 	if (csi2dev->flags & MXC_MIPI_CSI2_PM_POWERED)
@@ -700,8 +700,6 @@ static int mipi_csi2_pm_resume(struct device *dev)
 		return -EAGAIN;
 	}
 
-	if (csi2dev->running)
-		mipi_csi2_s_stream(sd, true);
 	csi2dev->flags |= MXC_MIPI_CSI2_PM_POWERED;
 	csi2dev->flags &= ~MXC_MIPI_CSI2_PM_SUSPENDED;
 
-- 
1.7.9.5

